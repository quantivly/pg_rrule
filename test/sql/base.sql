\set ECHO errors
BEGIN;
\i sql/pg_rrule.sql
\set ECHO all

SELECT 'FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z'::rrule;

SELECT get_byday('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=MO,TH,SU'::rrule);

SELECT get_freq('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z'::rrule);

SELECT * FROM
    unnest(
        get_occurrences('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=SA;BYHOUR=10;BYMINUTE=51;BYSECOND=2'::rrule,
            '2019-12-07 10:51:02+00'::timestamp with time zone)
    );

SELECT * FROM
    unnest(
        get_occurrences('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=SA;BYHOUR=10;BYMINUTE=51;BYSECOND=2'::rrule,
            '2019-12-07 10:51:02'::timestamp)
    );

WITH occurrences AS (
    SELECT unnest(
        get_occurrences(
           'FREQ=DAILY;BYHOUR=09;'::rrule,
            '2024-05-25 00:00:00'::timestamp,
            '2024-05-27 00:00:00'::timestamp
        )
    ) as occ
)
SELECT * FROM occurrences;

ROLLBACK;
