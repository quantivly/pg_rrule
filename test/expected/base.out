\set ECHO errors
SELECT 'FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z'::rrule;
               rrule                
------------------------------------
 FREQ=WEEKLY;UNTIL=20200101T045102Z
(1 row)

SELECT get_byday('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=MO,TH,SU'::rrule);
 get_byday 
-----------
 {2,5,1}
(1 row)

SELECT get_freq('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z'::rrule);
 get_freq 
----------
 WEEKLY
(1 row)

SELECT * FROM
    unnest(
        get_occurrences('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=SA;BYHOUR=10;BYMINUTE=51;BYSECOND=2'::rrule,
            '2019-12-07 10:51:02+00'::timestamp with time zone)
    );
WARNING:  Can't get timezone from current session! Fallback to UTC.
            unnest            
------------------------------
 Sat Dec 07 02:51:02 2019 PST
 Sat Dec 14 02:51:02 2019 PST
 Sat Dec 21 02:51:02 2019 PST
 Sat Dec 28 02:51:02 2019 PST
(4 rows)

SELECT * FROM
    unnest(
        get_occurrences('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=SA;BYHOUR=10;BYMINUTE=51;BYSECOND=2'::rrule,
            '2019-12-07 10:51:02'::timestamp)
    );
          unnest          
--------------------------
 Sat Dec 07 10:51:02 2019
 Sat Dec 14 10:51:02 2019
 Sat Dec 21 10:51:02 2019
 Sat Dec 28 10:51:02 2019
(4 rows)

WITH occurrences AS (
    SELECT unnest(
        get_occurrences(
           'FREQ=DAILY;BYHOUR=09;'::rrule,
            '2024-05-25 00:00:00'::timestamp,
            '2024-05-27 00:00:00'::timestamp
        )
    ) as occ
)
SELECT * FROM occurrences;
           occ            
--------------------------
 Sat May 25 09:00:00 2024
 Sun May 26 09:00:00 2024
(2 rows)

ROLLBACK;
